openlibrary/core/models.py

```diff
@@ -1153,7 +1147,7 @@ def create(
     ):
         """Creates a new Tag object."""
         current_user = web.ctx.site.get_user()
-        patron = current_user and current_user.username or 'ImportBot'
+        patron = current_user.get_username() if current_user else 'ImportBot'
```

Comment:
Deal with the case where `current_user.get_username() ` returns "" or None
```suggestion
        patron = current_user and current_user.get_username() or 'ImportBot'
```